### PR TITLE
Prevent accidentally dismissed admin notices

### PR DESCRIPTION
**Pull request by costasovo**
## Description

This PR fixes an issue where important notices might be dismissed by accident by any user and admin may miss important info.

## Code review notes

- The capability is `mailpoet_access_plugin_admin` rather than `manage_options`: editors can see these notices on MailPoet pages, so they must also be able to dismiss them.
- The approach mirrors the footer rating handler fix (STOMAIL-7609).

## QA notes

- On a MailPoet admin page with a dismissible MailPoet notice, dismissing it should still work and the notice should stay hidden after reload.
- A direct `POST` to `admin-ajax.php` with `action=dismissed_notice_handler` and no nonce should return HTTP 403 and leave the notice visible.

### How to test

1. Trigger a dismissible notice:
   ```bash
   pnpm wp eval 'MailPoet\DI\ContainerWrapper::getInstance()->get(MailPoet\Settings\SettingsController::class)->set("mailpoet_display_after_migration_notice", true);'
2. Open any MailPoet admin page (e.g. wp-admin/admin.php?page=mailpoet-homepage) — a green "Congrats! You're progressing well so far…" notice appears.
3. Dismiss it with the ✕ — the admin-ajax.php request returns 200 and the notice stays gone after reload.
4. Re-run step 1, then on any wp-admin page run in the browser console:
fetch(ajaxurl, {method: 'POST', body: new URLSearchParams({action: 'dismissed_notice_handler', type: 'mailpoet_display_after_migration_notice'})}).then((r) => console.log(r.status));
   Expect 403; the notice is still there after reload.


## Linked PRs

_N/A_

## Linked tickets

- [STOMAIL-8404](https://linear.app/a8c/issue/STOMAIL-8404/sec-2-missing-capability-nonce-on-notice-dismiss-ajax-15)

## After-merge notes

_N/A_

## Screenshots or screencast <!-- if applicable -->

_N/A_

## Tasks

- [x] I have added a changelog entry (`pnpm changelog:add --type=<type> --description=<description>`)
- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

## Preview

[Preview in WordPress Playground](https://account.mailpoet.com/playground/new/branch:fix-notice-dismiss-nonce)

_The latest successful build from `fix-notice-dismiss-nonce` will be used. If none is available, the link won't work._

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
No issues found.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

File: mailpoet/tests/integration/Util/Notices/PermanentNoticesTest.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+  public function testDismissibleNoticeMarkupContainsNonce() {
+    $wp = Stub::make(new WPFunctions, [
+      'wpCreateNonce' => Expected::once(function ($action) {
+        verify($action)->equals(Notice::DISMISS_NONCE_ACTION);
+        return 'test-nonce';
+      }),
+    ], $this);
+    $notice = new Notice(Notice::TYPE_WARNING, 'A message', '', 'some-notice-name', true, $wp);
```
[the remark this answers is not in the document]
Comment (costasovo) [reply mailpoet/tests/integration/Util/Notices/PermanentNoticesTest.php:84]: Added the `renderInParagraph = false` test case in 430589f673 — the nonce attributes are rendered separately in each branch, so both are now covered.

Skipping the `WPFunctions` routing: `escHtml__()` doesn't exist in the wrapper, and `lib/` calls `esc_html__`/`sanitize_text_field(wp_unslash())` directly in dozens of places, including the footer rating handler this PR mirrors. The behavioral calls (`currentUserCan`, `wpVerifyNonce`, `wpDie`, `wpCreateNonce`) already go through the wrapper and are stubbed in tests; escaping functions are pure and don't need replacing.

File: mailpoet/assets/js/src/dismissible-notice.jsx [hunk 2]
```diff
@@ -5,14 +5,13 @@ jQuery(($) => {
...
-        .closest('.mailpoet-dismissible-notice')
-        .data('notice');
+      const notice = $(this).closest('.mailpoet-dismissible-notice');
       $.ajax(window.ajaxurl, {
         type: 'POST',
         data: {
           action: 'dismissed_notice_handler',
-          type,
+          type: notice.data('notice'),
+          nonce: notice.data('nonce'),
```
Comment (lysyjan) [inline mailpoet/assets/js/src/dismissible-notice.jsx:17]: This introduces a 403 for notices that use the dismissible class but have no name. `SenderDomainAuthenticationNotices::display()` passes an empty `$dataNoticeName`, so `Notice::displayWPNotice()` emits neither `data-notice` nor `data-nonce`, and the click handler then POSTs `action=dismissed_notice_handler` with nothing else. On trunk the handler returned silently on the missing `type`; here the nonce check runs first and the request dies with "Security check failed."

Verified locally on wp-env: with MSS enabled and an unverified sender domain, every ✕ on the "Authenticate your sender domain" notice logs a 403 in the console. Nothing is persisted either way, so users won't notice, but it's a new error on a path that was clean before.

Could we skip the request when there is no notice name? A minimal version:

```js
const type = notice.data('notice');
if (!type) return;
```

Confirmed locally: with that guard the unnamed notice sends no request and the named ones still return 200 and stay hidden after reload. A small mocha spec for both cases would be worth adding, there is none for this file yet. Alternatively drop `mailpoet-dismissible-notice` from that notice's classes, your call.

File: mailpoet/lib/WP/Notice.php [hunk 3]
```diff
@@ -73,14 +80,14 @@ public function displayWPNotice() {
...
-        !empty($this->dataNoticeName) ? sprintf('data-notice="%s"', esc_attr($this->dataNoticeName)) : ''
+        !empty($this->dataNoticeName) ? sprintf('data-notice="%s" data-nonce="%s"', esc_attr($this->dataNoticeName), esc_attr($this->wp->wpCreateNonce(self::DISMISS_NONCE_ACTION))) : ''
       );
     } else {
       printf(
         '<div class="%1$s" %3$s>%2$s</div>',
         esc_attr($class),
         wp_kses_post($message),
-        !empty($this->dataNoticeName) ? sprintf('data-notice="%s"', esc_attr($this->dataNoticeName)) : ''
+        !empty($this->dataNoticeName) ? sprintf('data-notice="%s" data-nonce="%s"', esc_attr($this->dataNoticeName), esc_attr($this->wp->wpCreateNonce(self::DISMISS_NONCE_ACTION))) : ''
```
Comment (lysyjan) [inline mailpoet/lib/WP/Notice.php]: Optional/minor: the `data-notice="%s" data-nonce="%s"` expression is duplicated in both branches of `displayWPNotice()`. Computing it once before the `if` would also make it obvious that the nonce is emitted regardless of `renderInParagraph`.

File: mailpoet/assets/js/src/dismissible-notice.jsx (hunk 2 above)
Comment (costasovo) [reply mailpoet/assets/js/src/dismissible-notice.jsx:17]: Nice catch!  Addressed in https://github.com/mailpoet/mailpoet/pull/7082/commits/19a4e6c88a251cb4a26d4894b6c15e04b7f50bde following the recommendation.

File: mailpoet/lib/WP/Notice.php (hunk 3 above)
Comment (costasovo) [reply mailpoet/lib/WP/Notice.php]: Originally kept the branches because introducing a variable in sprintf triggers `phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped` and I wanted to avoid adding an inline ignore comment for this, because it can mask an issue in the future. In the end I went with a ternary operator in place to avoid both duplication and the ignore comment.

**Review by lysyjan — APPROVED** [review]
Verified locally on wp-env: the named notice sends `type` + `nonce`, returns `200` and stays hidden after reload; the unnamed sender-domain notice no longer fires a request; POST without nonce and as a subscriber both return `403` and leave the notice visible.
